service/history/decisionHandler.go

```diff
@@ -682,6 +690,12 @@ func (handler *decisionHandlerImpl) handleBufferedQueries(
 		return
 	}
 
+	// if its a heartbeat decision it means local activities may still be running on the worker
+	// which were started by an external event which happened before the query
+	if decisionHeartbeating {
+		return
```

Comment:
does this mean the queries are left to timeout or is it possible that after the local-activities are finished, these queries will be unblocked ?

Reply:
Yes its possible they will become answered. Basically a query can be answered on any decision that is created during the lifetime of it living in query registry.